common.gypi

```diff
@@ -24,6 +24,17 @@
 
     'openssl_fips%': '',
 
+    'v8_extra_library_files': [
+      './lib/v8_extras/ByteLengthQueuingStrategy.js',
```

Comment:
How are these files consumed by the build? I see that gypfiles in deps/v8 references `v8_extra_library_files`, are you relying on the compilation of v8 to grab these `lib/v8_extra` js files and compile them in? That will be more awkward for node-chakracore to work with.

Reply:
That's exactly what is happening. The V8 build compiles the v8_extras in. They are not loaded the same way as every other thing in core.

Reply:
can chakra not just put these files somewhere and eval them when a context is created? we do plenty of stuff elsewhere that chakra has to shim.

Reply:
Yes, chakra shims a lot of things, but so far it has been native APIs (or js APIs). Not side effects of the build system. Sure, it can be done, but I'd much prefer not to.

Personally I also feel like this makes the build of node more complex to reason about, it is that much less of a dependency tree and more of a dependency graph.

Reply:
If you wanted to have these JS files live in `lib` and be compiled to native, then my preferred approach would be to explicitly put that logic in node's build, not use a side effect of v8's build.

Reply:
i don't think it would really be feasible to rewrite this entire implementation to use commonjs... i'd be happy to write the pr for chakra to shim this.

Reply:
@MSLaguana are you saying to just put the list in node.gyp instead of common.gypi?

Reply:
Not the list, but the thing that consumes the list. The bit that takes the list of filenames and processes it to the point where the relevant build artifact is produced.

Reply:
Looking through the implementation, I'm not sure which aspects of it would be infeasible to implement as a normal module. Doing so would make it significantly easier to polyfill, would make it more likely that it could just be dropped in to `readable-stream`, and wouldn't require any changes to the build system or any coordination between v8 and chakra-core. @devsnek ... can you explain a bit more about why you think such a port wouldn't be feasible?

Reply:
v8 consumes the list and turns it into a c++ file that is linked internally to the engine. chakrashim would need to do something similar, calling those extras functions when new contexts are created.

Reply:
@jasnell because its a lot of stuff to rewrite (i don't have the time to do that) and i'd like to be able to collaborate with chromium by keeping at least similar implementations.